### PR TITLE
[functional tests] Adds a FILTER env var to filter logs in functional tests

### PR DESCRIPTION
**Pull request by tzakian**
This PR allows a user to set a custom filter over the logs that are produced by the functional tests when they fail  using a regex string defined in the `FILTER` env var.

e.g. setting
```
export FILTER="ABORTED|MISSING_DATA"
```
will only show lines in the logs that contain the words `ABORTED` or `MISSING_DATA`, all other lines will not be shown.

### CODE REVIEW

**Comment by sblackshear** [comment]
This is amazing, I can't wait to use it!

One question; will something like `FILTER=compiler PRETTY=1` work as expected?

**Comment by tzakian** [comment]
Yep! e.g.
```shell
>> PRETTY=1 FILTER=error cargo test different_currency
    Finished test [unoptimized + debuginfo] target(s) in 0.49s
     Running /Users/tzakian/work/libra/target/debug/deps/move_lang_functional_tests-f0a7128f3b5e19e3

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running /Users/tzakian/work/libra/target/debug/deps/functional_testsuite-1d7183245bbc1343

running 1 tests

====================================================================================================
tests/libra_account/different_currency.move

info: Evaluation Outputs
    [2] Error:

    error:

       ┌── /var/folders/hc/vmn1jkw963sfmhzltnnpqcym1lfkm6/T/.tmpKlGMXC:3:5 ───
       │
     3 │     1
       │     ^ Invalid return expression
       ·
     3 │     1
       │     - The type: integer
       ·
     2 │ fun main() {
       │     ---- Is not compatible with: '()'
       │





error: Unmatched Directives
    12 | // check: EXECUTED


error: Unmatched Errors
    [2] MoveSourceCompilerError("\u{1b}[0m\u{1b}[1m\u{1b}[38;5;9merror\u{1b}[0m\u{1b}[1m...

```

File: language/functional-tests/src/testsuite.rs [hunk 1]
```diff
@@ -54,6 +56,15 @@ fn pretty_mode() -> bool {
     pretty == "1" || pretty == "true"
 }
 
+fn print_stage(haystack: &str) -> bool {
```
Comment (vgao1996) [inline language/functional-tests/src/testsuite.rs:59]: "haystack"... Love this!

**Pull request closed without merging**